packages/phone-number-privacy/common/src/interfaces/requests.ts

```diff
@@ -150,35 +146,156 @@ export type DomainQuotaStatusRequest<
  * Options may be provided for authentication to prevent unintended parties from disabling a domain.
  *
  * @remarks Concrete request types are created by specifying the type parameters for Domain and
- * DomainOptions. If a DomainOptions type parameter is specified, then the options field is
+ * DomainOptions. If the specified Domain has associated options, then the options field is
  * required. If not, it must not be provided.
  */
-export type DisableDomainRequest<
-  D extends Domain = Domain,
-  O extends DomainOptions = D extends KnownDomain ? KnownDomainOptions<D> : never
-> = OmitIfNever<
-  {
-    /** Domain specification. Selects the PRF domain and rate limiting rules. */
-    domain: D
-    /** Domain-specific options. */
-    options: O
-    /** Client-specified session ID. */
-    sessionID: EIP712Optional<string>
-  },
-  'options'
->
+export type DisableDomainRequest<D extends Domain = Domain> = {
+  /** Domain specification. Selects the PRF domain and rate limiting rules. */
+  domain: D
+  /** Domain-specific options. */
+  options: DomainOptions<D>
+  /** Client-specified session ID. */
+  sessionID: EIP712Optional<string>
+}
 
 /** Union type of Domain API requests */
-export type DomainRequest<
-  D extends Domain = Domain,
-  O extends DomainOptions = D extends KnownDomain ? KnownDomainOptions<D> : never
-> =
-  | DomainRestrictedSignatureRequest<D, O>
-  | DomainQuotaStatusRequest<D, O>
-  | DisableDomainRequest<D, O>
+export type DomainRequest<D extends Domain = Domain> =
+  | DomainRestrictedSignatureRequest<D>
+  | DomainQuotaStatusRequest<D>
+  | DisableDomainRequest<D>
+
+// NOTE: Next three functions are a bit repetitive. An attempt was made to combine them, but the
+// type signature got quite complicated. Feel free to attempt it if you are motivated.
```

Comment:
```suggestion
// type signature got quite complicated. Feel free to attempt it if you are motivated. TODO(Alec)
```
Leaving this reminder here for myself